kafka-connector/src/main/java/com/google/pubsub/kafka/source/CloudPubSubSourceConnector.java

```diff
@@ -193,6 +198,30 @@ public ConfigDef config() {
             ConfigDef.Range.between(1, Integer.MAX_VALUE),
             Importance.MEDIUM,
             "The minimum number of messages to batch per pull request to Cloud Pub/Sub.")
+        .define(
+            CPS_STREAMING_PULL_ENABLED,
+            Type.BOOLEAN,
+            false,
+            Importance.MEDIUM,
+            "Whether to use streaming pull for the connector to connect to Cloud Pub/Sub. If provided, cps.maxBatchSize is ignored.")
+        .define(
+            CPS_STREAMING_PULL_PARALLEL_PULL_COUNT,
+            Type.INT,
+            10,
```

Comment:
Is it best to achieve parallelism through the parallel pull count or by increasing the number of tasks. Having a high parallel pull count can result in worse performance at times, I believe due to contention. We default the client to a single stream and at the very least, I think we should keep that the default. But we should also consider if we even need this knob given the ability to control the number of tasks.

Reply:
I worry that it will be high overhead to have a bunch of 5MiB/s tasks? But I can remove this if task fanout is the preferred way to scale.

Reply:
Removed.

Reply:
I think we can hit up to 10MB/s per stream. So let's leave this out for now as you have, but we can always add it in later if the overhead of each task is too much and the bottleneck in receiving more messages is the single stream.